overlord/configstate/configcore/picfg.go

```diff
@@ -89,10 +97,14 @@ func updatePiConfig(path string, config map[string]string) error {
 
 func piConfigFile(opts *fsOnlyContext) string {
 	rootDir := dirs.GlobalRootDir
+	subdir := "/boot/uboot"
 	if opts != nil {
 		rootDir = opts.RootDir
+		if opts.UC20 {
+			subdir = ""
+		}
 	}
```

Comment:
I forgot but we will have separately to also make this code do the right thing under UC20 when invoked via Run which atm is based on setting opts to nil

Reply:
yes, now the right thing is done here (but I broke out the main behavior here instead for easier review: https://github.com/snapcore/snapd/pull/10088)